(#3) Display Listing Results for Disease

sarinapadilla opened the pull request:
* Closes #3
* Update index.js with API endpoints and requestInterceptors
* Update index.html with initialize fields for Disease
* Add queryParamType constant
* Update axios-client to handle multiple APIs and endpoints
* Add tests for updated axios setup
* Add cleanURI util and test
* Update getClinicalTrials action test
* Add getListingInformationById service
* Add getListingInformationByName service
* Add tests for getListingInformationByName/Id services
* Update app tests to include multiple endpoints
* Update HoC to perform Listing API fetch before loading results page
* Add matchQueryParam util and tests
* Add unit tests for HoC
* Add disease dynamic route
* Add codeOrPurlPath to routing
* Add Disease component
* Display intro text and listing results in Disease
* Add unit tests for Disease
* Add mock data for Disease integration tests
* Add integration tests for Disease component
* Update NoParamsPageNotFound tests with correct URL
* Fix syntax for onSubmit in PageNotFound form
* Fix linter errors

sarinapadilla commented:
@VictoriaSunNIH @mworrest @blilianyu here is the link for testing: https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-36/{id-or-pretty-url-name}?cfg=0
?cfg=0 is necessary because that is the config that determines it is a Disease listing page.

Reminder that you need to be connected to VPN for testing (since the Listing API is still on webapis-dev)!

You can replace {id-or-pretty-url-name} with a single c-code or comma-separated list of c-codes, but the codes must all associate with one disease name or the page will error. (There is another story for handling that error page.) You can also replace it with a pretty URL name. Try different combinations: use a single c-code from a list of multiple for one disease, all the c-codes for one disease out of the order in the file, codes from multiple diseases (to check that it errors), pretty URL names, etc.

To find c-code combinations or pretty URL names to test, I suggest using the Listing API dump to get some values to put in the URL.
[ListingInfo.txt](https://github.com/NCIOCPL/clinical-trials-listing-app/files/6123363/ListingInfo.txt)
Make sure you're using a code or pretty URL name for a disease - they're a little further down in the file.

If you need any help finding values for testing, just let me know and I'm happy to assist! :)

KateMashkinaNIH reviewed (changes requested):
The analytics test is missing for disease

KateMashkinaNIH commented:
og:title on prod reads the following <meta property="og:title" content="Disease Dynamic Trial Listing Page"> - are we matching prod, or is there a new requirement (AC in the ticket is different)?

sarinapadilla commented:
@KateMashkinaNIH Good looks, I've added a Disease analytics test. For the og:title, the AC in the ticket is correct. I spoke to @lburack and the og:title on Prod is a remnant of overlooked meta tag values in the old dynamic listing pages, so we updated the value to be more reflective of the page.

alinai commented:
Should I expect to see "v?" in the url when I hover on Title link? On hovering a title link on Manual Listing pages, it displays the Trial Description page with the url https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/v?id=NCI-2019-00744. Should DTLP also show us that kind of url on hovering title link?

sarinapadilla commented:
@alinai The results item links weren't being set up correctly because I hadn't pulled in Alex's recent changes. Those have been pulled in and now all of the links should point to /clinical-trials/{nci_id} as per requirements.

VictoriaSunNIH commented:
@sarinapadilla  - this looks good to me.  Test pretty URLs, individual c-codes, c-code combinations and c-codes combinations that didn't exist.  And thanks for pulling in Alex's changes so we could see the links working correctly!

alinai commented:
Looks good to me. Tested different combinations with a single c-code, multiple c-codes (C3099,C7955,C7956") for one disease, all the c-codes for one disease, by changing c-codes from multiple c-codes to check that we get error page, with pretty URL names and with c-codes that return no results page. Thanks @sarinapadilla!

seyilonge-nci commented on src/services/api/actions/getListingInformationById.js:
```diff
@@ -0,0 +1,20 @@
+ * Gets a listing information from the Listing Information API matching the requested concept ID(s)
+ *
+ * @param {Object} params the parameters to fetch
+ * @param {string[]} params.queryParam the concept ID(s) to match
```
Function takes only one param, this does not match function signature. Could you correct this please.

sarinapadilla replied:
Fixed!

seyilonge-nci commented on src/services/api/actions/getListingInformationByName.js:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Gets a listing information from the Listing Information API matching the requested pretty URL name
+ *
+ * @param {Object} params the parameters to fetch
```
This doesn't seem to be a param in the function. Similar to the issue above.

sarinapadilla replied:
Fixed!
